cypress/integration/focus-trap-demo.spec.js

```diff
@@ -55,4 +54,73 @@ describe('<FocusTrap> component', () => {
     cy.focused().should(([nextFocusedEl]) =>
       expect(nextFocusedEl).not.equal(previousFocusedEl));
   });
+
+  it('On trap mounts and activates, focus on manually specified input element', () => {
+    cy.get('#demo-ffne')
+      .as('testRoot');
+
+    // activate trap
+    cy.get('@testRoot').findByRole('button', { name: 'activate trap' })
+      .click();
+
+    // instead of next tab-order element being focused, element specified should be focused
+    cy.get('@testRoot').findByRole('textbox', { name: 'Initially focused input' }).should('be.focused');
+  });
+
+  it('Escape key does not deactivate trap. Instead, click on "deactivate trap" to deactivate trap', () => {
+    cy.get('#demo-ffne')
+      .as('testRoot');
+
+    // activate trap
+    cy.get('@testRoot').findByRole('button', { name: 'activate trap' }).as('lastlyFocusedElementBeforeTrapIsActivated')
+      .click();
+
+    // trying deactivate trap by ESC
+    cy.get('@testRoot').findByRole('textbox', { name: 'Initially focused input' }).as('trapChild').focus().type('{esc}');
+
+    // ESC does not deactivate the trap
+    cy.get('@trapChild').should('exist').should('be.focused');
+
+    // click on deactivate trap button to deactivate trap
+    cy.get('@testRoot').findByRole('button', { name: 'deactivate trap' }).click();
+    cy.get('@trapChild').should('not.exist');
+    cy.get('@lastlyFocusedElementBeforeTrapIsActivated').should('be.focused');
+  });
+
+  it('Can be deactivated while staying mounted  ', () => {
+    cy.get('#demo-special-element')
+      .as('testRoot');
+
+    // activate trap
+    cy.get('@testRoot').findByRole('button', { name: 'activate trap' }).as('lastlyFocusedElementBeforeTrapIsActivated')
+      .click();
+
+    // stay mounted after deactivation
+    cy.get('@testRoot').findByRole('button', { name: 'deactivate trap' }).as('trapChild').click();
+    cy.get('@lastlyFocusedElementBeforeTrapIsActivated').should('be.focused');
+    cy.get('@trapChild').should('exist');
+  });
+
+  it('Can click outside of trap to deactivate and click carries through  ', () => {
+    cy.get('#demo-special-element')
+      .as('testRoot');
+
+    // activate trap
+    cy.get('@testRoot').findByRole('button', { name: 'activate trap' }).click();
+
+    // click outside to deactivate trap and also click carries through
+    cy.findByRole('button', { name: 'pass thru click' }).click().should('be.focused');
+    cy.get('@testRoot').findByText('Clicked!');
+  });
+
+  it('Focus on element with "autofocus" attribute than the 1st tab order element within mounted trap children', () => {
+    cy.get('#demo-autofocus')
+      .as('testRoot');
+
+    // activate trap
+    cy.get('@testRoot').findByRole('button', { name: 'activate trap' }).click();
+
+    // element with "autofocus" attribute is focused
+    cy.findByTestId('autofocus-el').should('be.focused');
```

Comment:
demo page is updated a little to have some [easier selector](https://github.com/focus-trap/focus-trap-react/pull/103/files#diff-d7e657e37f7e3c3155b02d20d9560b6aR41) for e2e test.